src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,159 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int quantity = 0;
+    private Node<T> node;
+    private Node<T> temp;
+
     @Override
     public void add(T value) {
+        rewindNodeToHead();
+        if (quantity == 0) {
+            node = new Node(null, value, null, true, true);
```

Comment:
When creating a new Node<T>, pass all the fields to the Node constructor. Ensure that you are passing the correct parameters to the Node constructor.